feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/vif.c

```diff
@@ -495,6 +504,18 @@ static void vif_config_custom_opt_set(struct blob_buf *b,
 			blobmsg_add_string(b, NULL, operator_name);
 			blobmsg_close_array(b, n);
 		}
+		else if (strcmp(opt, "dynamic_vlan") == 0) {
+			blobmsg_add_string(b, "dynamic_vlan", value);
+
+			if (strcmp(value, "1") == 0 || strcmp(value, "2") == 0) {
+				strncpy(value, "/etc/hostapd.vlan", 20);
+				blobmsg_add_string(b, "vlan_file", value);
+				strncpy(value, "0", 20);
+				blobmsg_add_string(b, "vlan_naming", value);
+				strncpy(value, "br-wan.", 20);
```

Comment:
Do you really want the '.' in 'br-wan.'?  Just checking.

Reply:
Yes, the hostapd then creates a bridge called br-wan.<vlan_id> acting as a VLAN bridge.